### PR TITLE
Fix preprocess tsconfig

### PR DESCRIPTION
**Pull request by csjh**
### Description

Fixes preprocess tsconfig to include `*.cjs` and `*.js`, and fixes all type errors that popped up as a result

### Checklist

- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

**Comment by csjh** [comment]
Not 100% sure about one of the changes here, `prismjs` -> `prismjs/components` (base `prismjs` doesn't have the `alias` field)

@ItsMeBrianD do you remember why [this change](https://github.com/evidence-dev/evidence/commit/04618d84e0132b29fb8da91b14906b0807b115e7#diff-22d1d9031151b0fb01e1853325afcb96542db330dce58d128e190001614ded0e) was made before?

File: packages/lib/preprocess/src/extract-queries/extract-queries.cjs [hunk 1]
```diff
@@ -135,13 +138,15 @@ const strictBuild = process.env.VITE_BUILD_STRICT === 'true';
...
+ * @param {string} content
  * @returns {Query[]}
  */
 const extractQueries = (content) => {
+	/** @type {Query[]} */
 	const queries = [];
 
-	queries.push(...extractExternalQueries(content));
+	// todo: second parameter is filename but we don't have that here?
+	queries.push(...extractExternalQueries(content, ''));
```
Comment (zachstence) [inline packages/lib/preprocess/src/extract-queries/extract-queries.cjs:149]: You can make the second argument `filename` optional in the JSDoc comment on `extractExternalQueries`
https://stackoverflow.com/a/29441515/9080819

File: packages/lib/preprocess/src/extract-queries/extract-queries.cjs [hunk 2]
```diff
@@ -73,10 +78,7 @@ const extractExternalQueries = (content, filename) => {
 		return true;
 	};
 
-	/**
-	 * @type Query[]
-	 */
-	return frontmatter.queries
+	return /** @type {string[]} */ (frontmatter.queries)
```
Comment (zachstence) [inline packages/lib/preprocess/src/extract-queries/extract-queries.cjs:81]: There's a `.map` statement below with two branches, one for `string` and `object`, so I don't think this type is correct: We don't expect everything in this array to be strings.

**Review by zachstence — CHANGES_REQUESTED** [review]

**Review by ItsMeBrianD — DISMISSED** [review]
Zach's comments

File: packages/lib/preprocess/src/extract-queries/extract-queries.cjs (hunk 1 above)
Comment (csjh) [reply packages/lib/preprocess/src/extract-queries/extract-queries.cjs:149]: I'm more wondering if we should be passing a file name into `extractQueries`. Not sure if we used to pass a file name in for the better error in `extractExternalQueries`

Comment (zachstence) [reply packages/lib/preprocess/src/extract-queries/extract-queries.cjs:150]: Maybe, maybe not. But for now we should just maintain the argument being optional with the added types

**Review by zachstence — APPROVED** [review]

**Comment by zachstence** [comment]
Needs approval from @ItsMeBrianD

**Review by ItsMeBrianD — APPROVED** [review]

**Pull request merged**